agent/lm_agent/workload_managers/slurm/reservations.py

```diff
@@ -0,0 +1,132 @@
+"""Slurm reservation CRUD module."""
+import shlex
+from typing import Union
+
+from lm_agent.config import settings
+from lm_agent.logs import logger
+from lm_agent.utils import run_command
+from lm_agent.workload_managers.slurm.common import SCONTROL_PATH
+
+
+async def scontrol_create_reservation(licenses: str, duration: str) -> bool:
+    """
+    Create the reservation for licenses with the specified duration.
+
+    Duration format: [days-]hours:minutes:seconds
+    Licenses format: foo.bar@server:123,foo.baz@server:234
+
+    Returns True if the reservation was created successfully, otherwise returns False.
+    """
+    cmd = [
+        SCONTROL_PATH,
+        "create",
+        "reservation",
+        f"user={settings.LM_USER}",
+        f"reservation={settings.RESERVATION_IDENTIFIER}",
+        "starttime=now",
+        f"duration={duration}",
+        "flags=license_only",
+        f"licenses={licenses}",
+    ]
+
+    logger.debug(f"#### Creating reservation for {licenses} with duration {duration} ####")
+    reservation_output = await run_command(shlex.join(cmd))
+
+    logger_error_message = f"#### Failed to create reservation {settings.RESERVATION_IDENTIFIER} ####"
+    if not reservation_output:
+        logger.error(logger_error_message)
+        return False
+    if isinstance(reservation_output, str):
+        if f"Reservation created: {settings.RESERVATION_IDENTIFIER}" not in reservation_output:
```

Comment:
this is bad. I had problems with it to debug the lm agent charm. Is there a way to check the return code?

Reply:
I updated the `run_command` function to to as @dusktreader suggested.
I'll update the logic in the methods taking this into consideration.

Reply:
@matheushent now the run_command function will raise an exception if the return code != 0.
This made possible to remove the output checking here.
Is it better now?

Reply:
@julianaklulo I'm a little bit confused because I see the try-except blocks but I couldn't find the change in the function in this PR. I also see some checks by strings ([here](https://github.com/omnivector-solutions/license-manager/pull/188/commits/15aefb998d938dd0fec1b504cfd47ad11e3dca7a#diff-b969ae58520282e57a72883f59250a6cc46181fc9cbe4abd03c08c8faa7513d1R100) for example)

Reply:
I removed all checks by strings in [15aefb9](https://github.com/omnivector-solutions/license-manager/commit/15aefb998d938dd0fec1b504cfd47ad11e3dca7a). Please look at the files changed to see the new version of the code.

Reply:
The modification in the run_command function to raise the exception is already in main [32d35f7](https://github.com/omnivector-solutions/license-manager/commit/32d35f7f7b7111a8e1ce10b682480b03a5317641), it was done in #187 that had been opened before this PR.